Serverless/Serverless computing.md

```diff
@@ -0,0 +1,29 @@
+As cloud services became more and more popular, they gave way to a new computing execution model: **serverless**. In this model the cloud provider allocates the necessary resources to run the code we want to run, taking care of everything behind the scenes and allowing the developers to focus more on the development of the business logic.
+
+The term "serverless" might look like a misnomer at a first glance - our code is, after all, executed on some kind of a server - but its point is to describe the developer's perspective. Using this model, the developers and DevOps engineers don't have to be concerned with all the usual server maintenance stuff like allocating CPU and memory, managing the network and its security, and planning the capacity, nor do they have to worry about having more resources than the application requires at a specific point in time. The application uses the resources provided by the cloud service only while it is running, meaning that it can be scaled up or down when needed, or be completely shut down when not in use.
+
+The execution of our code can be triggered by various triggers like an HTTP request, a queue message, changes one the blob storage, or time triggers. All of them will have the same underlying infrastructure and setup, with the only difference being the entry point. 
+
+The main benefits of using serverless solutions are:
+
+- **Writing less code** - The developer can focus on the business logic implementation and utilize the ready-to-use infrastructure features.
+- **Maintaining less infrastructure** - Cloud infrastructure providers manage servers, security, and scaling
+- **Saving on costs** - No need to buy expensive machines and infrastructure, pay only for the resources that were used by our application.
+- **Scalability** - Our services automatically scale up or down depending on usage.
+
+Even though the benefits might sound like a dream come true, before you start serverless-ing everything from your APIs to your toaster, consider some of the disadvantages of this approach:
```

Comment:
Would you consider adding something like `State management` also to the list of disadvantages? I know that today the stateless approach is much more popular. However, I think that we should mention that maintaining state between invocations can require additional effort for someone who wants to use serverless and maintain state, although in that case I would probably avoid serverless altogether.

Reply:
I see your point, and I do agree that state management would require additional effort, but mentioning that would be a bit contradictory to the best practice point in the About Azure Function section: "_Functions should be **stateless** and **idempotent** if possible._"

While I could mention something like "Serverless functions should be stateless, but if we do need to keep track of the state, that will require additional effort", I find it a bit weird to mention that not following a best practice would not be good.

Do you think I could move that best practice point to this chapter since it could be applicable to all serverless apps?

Reply:
I think that's a good idea, to move that best practice point to this chapter :)